### ISSUE TITLE
FNX3-16743 ⁃ [Bug] Sign in on a fresh install (still) crashes the app

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#3031]
I'm filling this as a **regression** of https://github.com/mozilla-mobile/fenix/issues/1975 and I've verified that my build also has the fix for the other 'Display already acquired' (https://github.com/mozilla-mobile/fenix/pull/3023).

## Steps to reproduce
1. Go to settings.
1. Click sign in, actually sign in.

### Expected behavior
* Don't crash.

### Actual behavior
* Crashes

```
2019-05-30 21:23:44.464 1067-1067/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.debug, PID: 1067
    java.lang.RuntimeException: Failed to call observer method
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:226)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:194)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:185)
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36)
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:361)
        at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:300)
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:339)
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:145)
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:131)
        at androidx.fragment.app.Fragment.performStart(Fragment.java:2675)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:908)
        at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManagerImpl.java:2087)
        at androidx.fragment.app.FragmentManagerImpl.postponePostponableTransactions(FragmentManagerImpl.java:1952)
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1862)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1817)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1717)
        at androidx.fragment.app.FragmentManagerImpl$2.run(FragmentManagerImpl.java:150)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7116)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:925)
     Caused by: java.lang.IllegalStateException: Display already acquired
        at org.mozilla.geckoview.GeckoSession.acquireDisplay(GeckoSession.java:2181)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:390)
        at org.mozilla.geckoview.GeckoView.setSession(GeckoView.java:353)
        at mozilla.components.browser.engine.gecko.GeckoEngineView.render(GeckoEngineView.kt:84)
        at mozilla.components.feature.session.EngineViewPresenter.renderSession$feature_session_release(EngineViewPresenter.kt:51)
        at mozilla.components.feature.session.EngineViewPresenter.start(EngineViewPresenter.kt:31)
        at mozilla.components.feature.session.SessionFeature.start(SessionFeature.kt:27)
        at mozilla.components.support.base.feature.ViewBoundFeatureWrapper.start$support_base_release(ViewBoundFeatureWrapper.kt:157)
        at mozilla.components.support.base.feature.LifecycleBinding.start(ViewBoundFeatureWrapper.kt:193)
        at java.lang.reflect.Method.invoke(Native Method)
        at androidx.lifecycle.ClassesInfoCache$MethodReference.invokeCallback(ClassesInfoCache.java:216)
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeMethodsForEvent(ClassesInfoCache.java:194) 
        at androidx.lifecycle.ClassesInfoCache$CallbackInfo.invokeCallbacks(ClassesInfoCache.java:185) 
        at androidx.lifecycle.ReflectiveGenericLifecycleObserver.onStateChanged(ReflectiveGenericLifecycleObserver.java:36) 
        at androidx.lifecycle.LifecycleRegistry$ObserverWithState.dispatchEvent(LifecycleRegistry.java:361) 
        at androidx.lifecycle.LifecycleRegistry.forwardPass(LifecycleRegistry.java:300) 
        at androidx.lifecycle.LifecycleRegistry.sync(LifecycleRegistry.java:339) 
        at androidx.lifecycle.LifecycleRegistry.moveToState(LifecycleRegistry.java:145) 
        at androidx.lifecycle.LifecycleRegistry.handleLifecycleEvent(LifecycleRegistry.java:131) 
        at androidx.fragment.app.Fragment.performStart(Fragment.java:2675) 
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManagerImpl.java:908) 
        at androidx.fragment.app.FragmentManagerImpl.addAddedFragments(FragmentManagerImpl.java:2087) 
        at androidx.fragment.app.FragmentManagerImpl.postponePostponableTransactions(FragmentManagerImpl.java:1952) 
        at androidx.fragment.app.FragmentManagerImpl.executeOpsTogether(FragmentManagerImpl.java:1862) 
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1817) 
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1717) 
        at androidx.fragment.app.FragmentManagerImpl$2.run(FragmentManagerImpl.java:150) 
        at android.os.Handler.handleCallback(Handler.java:883) 
        at android.os.Handler.dispatchMessage(Handler.java:100) 
        at android.os.Looper.loop(Looper.java:214) 
        at android.app.ActivityThread.main(ActivityThread.java:7116) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:925) 
```

### Device information
* Android device: Pixel (Android Q) & Pixel 2 (Android 9)
* Fenix version: ab39a508

**Comment by vesta0** [comment]
We haven't landed support for Android Q yet :)

**Comment by jonalmeida** [comment]
This is reproducible on all devices, I didn't have any more devices left that didn't have sync installed on it at the time. :)

**Comment by colintheshots** [comment]
This appears to be fixed with PR #3043

**Comment by sv-ohorvath** [comment]
Can't reproduce it with the STR described, I'll consider this as fixed.
